src/parser.rs

```diff
@@ -114,14 +114,21 @@ fn parse_err(args: &[u8]) -> ControlOp {
     ControlOp::Err(err_description.to_string())
 }
 
-pub(crate) fn expect_info(reader: &mut BufReader<TcpStream>) -> io::Result<ServerInfo> {
-    let op = parse_control_op(reader)?;
+pub(crate) fn expect_info(reader: &mut TcpStream) -> io::Result<ServerInfo> {
+    let mut buf = Vec::with_capacity(512);
```

Comment:
This just initial capacity yes? Grows on demand if larger?

Reply:
Right, it will copy things to a larger buffer if it runs out. I saw that on my laptop with a very simple configuration it was like 283 bytes. Do you have a sense for how large this can be in larger production environments?

Reply:
Mostly influenced by how many servers in the cluster. So if you set up a cluster with say 50 nodes (We do not recommend this BTW) you will see a larger INFO.